zabbix/repo.sls

```diff
@@ -53,6 +53,16 @@
     - gpgcheck: 1
     - gpgkey: https://repo.zabbix.com/RPM-GPG-KEY-ZABBIX-79EA5ED4
 
+{%- elif salt['grains.get']('os_family') == 'Suse' %}
+zabbix_repo:
```

Comment:
Missed this part, thanks @myii for pointing this issue. Variable must be used in state id to prevent duplicatees.
```suggestion
{{ id_prefix }}_repo:
```

Reply:
@hatifnatt Yes, unfortunately that's not enough, getting some strange errors after fixing that:

* https://gitlab.com/myii/zabbix-formula/-/jobs/1186085485
* https://gitlab.com/myii/zabbix-formula/-/jobs/1186085486

```sls
       [ERROR   ] An un-handled exception was caught by salt's global exception handler:
       StopIteration: 
       Traceback (most recent call last):
         File "/usr/bin/salt-call", line 8, in <module>
           sys.exit(salt_call())
         File "/usr/lib/python3.8/site-packages/salt/scripts.py", line 449, in salt_call
           client.run()
         File "/usr/lib/python3.8/site-packages/salt/cli/call.py", line 58, in run
           caller.run()
         File "/usr/lib/python3.8/site-packages/salt/cli/caller.py", line 112, in run
           ret = self.call()
         File "/usr/lib/python3.8/site-packages/salt/cli/caller.py", line 219, in call
           ret["return"] = self.minion.executors[fname](
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 1235, in __call__
           return self.loader.run(run_func, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2268, in run
           return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2283, in _run_as
           return _func_or_method(*args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/executors/direct_call.py", line 12, in execute
           return func(*args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 1235, in __call__
           return self.loader.run(run_func, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2268, in run
           return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2283, in _run_as
           return _func_or_method(*args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/modules/state.py", line 1115, in highstate
           ret = st_.call_highstate(
         File "/usr/lib/python3.8/site-packages/salt/state.py", line 4527, in call_highstate
           return self.state.call_high(high, orchestration_jid)
         File "/usr/lib/python3.8/site-packages/salt/state.py", line 3251, in call_high
           high, ext_errors = self.reconcile_extend(high)
         File "/usr/lib/python3.8/site-packages/salt/state.py", line 1627, in reconcile_extend
           state_type = next(x for x in body if not x.startswith("__"))
       StopIteration
       Traceback (most recent call last):
         File "/usr/bin/salt-call", line 8, in <module>
           sys.exit(salt_call())
         File "/usr/lib/python3.8/site-packages/salt/scripts.py", line 449, in salt_call
           client.run()
         File "/usr/lib/python3.8/site-packages/salt/cli/call.py", line 58, in run
           caller.run()
         File "/usr/lib/python3.8/site-packages/salt/cli/caller.py", line 112, in run
           ret = self.call()
         File "/usr/lib/python3.8/site-packages/salt/cli/caller.py", line 219, in call
           ret["return"] = self.minion.executors[fname](
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 1235, in __call__
           return self.loader.run(run_func, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2268, in run
           return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2283, in _run_as
           return _func_or_method(*args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/executors/direct_call.py", line 12, in execute
           return func(*args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 1235, in __call__
           return self.loader.run(run_func, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2268, in run
           return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/loader.py", line 2283, in _run_as
           return _func_or_method(*args, **kwargs)
         File "/usr/lib/python3.8/site-packages/salt/modules/state.py", line 1115, in highstate
           ret = st_.call_highstate(
         File "/usr/lib/python3.8/site-packages/salt/state.py", line 4527, in call_highstate
           return self.state.call_high(high, orchestration_jid)
         File "/usr/lib/python3.8/site-packages/salt/state.py", line 3251, in call_high
           high, ext_errors = self.reconcile_extend(high)
         File "/usr/lib/python3.8/site-packages/salt/state.py", line 1627, in reconcile_extend
           state_type = next(x for x in body if not x.startswith("__"))
       StopIteration
```

Reply:
@ghormoon @hatifnatt Looks like these `extend` blocks need to handle `Suse` as well?

* https://github.com/saltstack-formulas/zabbix-formula/search?q=extend+filename%3Arepo.sls

Reply:
@ghormoon I see what the problem is in the `repo.sls` files (the `extends` haven't been set correctly).  Just trying something in my fork and will link inline with the results.

Reply:
Yes there is a problem with extend
https://gitlab.com/myii/zabbix-formula/-/jobs/1186085485#L1687-#L1691 it's rendering as simple `{}` that's does not look like correct yaml.

Reply:
@hatifnatt Actually, the current code isn't even hitting the `{}`, it's actually splitting the `RedHat` block with a `Suse` block.  I've mentioned the fixes below.